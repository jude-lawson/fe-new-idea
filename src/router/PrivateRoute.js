import React from 'react';
import { Route, Redirect } from 'react-router-dom';

export const PrivateRoute = ({ authenticated, component: Component, ...rest }) => console.log(rest)||(
  <Route {...rest} render={props => (
    authenticated ?
      (
        <Component {...props} />
      ) :
      (
        <Redirect to="/" />
      )
  )} />
);