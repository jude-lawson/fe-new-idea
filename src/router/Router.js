import React from 'react';
import PropTypes from 'prop-types';
import { Router, Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import createHistory from 'history/createBrowserHistory';

import App from '../Components/App/App';
import Profile from '../Containers/Profile/Profile';
import { PrivateRoute } from './PrivateRoute';

export const history = createHistory();

export const AppRouter = ({ authenticated }) => (
  <Router history={history}>
    <Switch>
      <Route path="/" exact component={App}/>
      <PrivateRoute authenticated={authenticated} path="/profile" exact  component={Profile} />
    </Switch>
  </Router>
);

AppRouter.propTypes = {
  authenticated: PropTypes.string
};

export const mapStateToProps = state => ({
  authenticated: state.user.id
});

export default connect(mapStateToProps)(AppRouter);