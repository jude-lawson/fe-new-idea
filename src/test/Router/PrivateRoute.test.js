import React from 'react';
import { shallow } from 'enzyme'
import { PrivateRoute } from '../../router/PrivateRoute';

describe('<AppRouter/>', () => {
  let wrapper;
  const mockComponent = {};
  const mockProps = { path: '/' };

  beforeEach(() => wrapper =
  shallow(
    <PrivateRoute
      authenticated={'1111'}
      component={mockComponent}
      props={mockProps}
    />
    ))

  test('should render when no user authenticated', () => {
    wrapper = shallow(<PrivateRoute />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render when user is authenticated', () => {
    expect(wrapper).toMatchSnapshot()
  });
});
