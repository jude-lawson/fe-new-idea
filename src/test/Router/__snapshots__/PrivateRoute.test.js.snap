// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter/> should render when no user authenticated 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PrivateRoute />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;

exports[`<AppRouter/> should render when user is authenticated 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PrivateRoute
    authenticated="1111"
    component={Object {}}
    props={
        Object {
            "path": "/",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "props": Object {
        "path": "/",
      },
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "props": Object {
          "path": "/",
        },
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
